### PR TITLE
Feature/user roles

### PR DESCRIPTION
**Pull request by kmih**

### CODE REVIEW

File: src/main/java/com/epam/idea/core/service/impl/IdeaServiceImpl.java [hunk 1]
```diff
@@ -83,21 +93,34 @@ public Idea update(final long ideaId, final Idea source) {
...
 		final List<Idea> ideas = ideaRepository.findByTagId(tagId);
-		ideas.forEach(idea -> Hibernate.initialize(idea.getRelatedTags()));
+		ideas.forEach(idea -> {
+			Hibernate.initialize(idea.getRelatedTags());
+			if (isAnonymous) {
+				idea.setLiked(isCurrentUserLikedIdea(idea.getId()));
+			}
+		});
 		return ideas;
 	}
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/core/service/impl/IdeaServiceImpl.java:107]: May be, it makes sense to move logic "boolean isAnonymous = Securi... if (isAnonymous) ...idea.setLiked(..." to separate method?

Comment (kmih) [inline src/main/java/com/epam/idea/core/service/impl/IdeaServiceImpl.java:107]: Done

**Comment by Hanna-Aliakseichykava** [comment]
:+1:

**Comment by LostFan** [comment]
:+1:

**Pull request merged**